### PR TITLE
fix: plugin delete lifecycle method will not be triggered when the plugin is uninstalled

### LINKED ISSUE #4238: 插件的 delete 生命周期方法在卸载时没有被触发
### What is version of Halo has the issue?

2.7.0

### What database are you using?

H2

### What is your deployment method?

Docker

### Your site address.

_No response_

### What happened?

如题，当插件被卸载时 delete 生命周期方法没有被触发

### Relevant log output

_No response_

### Additional information

/kind bug
/area core
/milestone 2.8.0
/assign

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind bug
/kind improvement
/area core
/milestone 2.8.x

#### What this PR does / why we need it:
修复插件被卸载时 delete 生命周期方法不会被触发的问题

how to test it?
1. 测试开发模式下卸载插件，delete 生命周期方法被触发且不会误删项目目录
2. 测试生产模式下插件卸载，文件正确被删除且触发 delete 生命生命周期方法

#### Which issue(s) this PR fixes:
Fixes #4238

#### Does this PR introduce a user-facing change?
```release-note
修复插件被卸载时 delete 生命周期方法不会被触发的问题
```

### CODE REVIEW

File: application/src/test/java/run/halo/app/plugin/DefaultDevelopmentPluginRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Tests for {@link DefaultDevelopmentPluginRepository}.
+ *
+ * @author guqing
+ * @since 2.7.0
+ */
+class DefaultDevelopmentPluginRepositoryTest {
+
+    private PluginRepository developmentPluginRepository;
+    private Path tempDir;
```
Comment (JohnNiang) [inline application/src/test/java/run/halo/app/plugin/DefaultDevelopmentPluginRepositoryTest.java:22]: 可直接用 `@TempDir` 替代手动创建和清理。

**Review by JohnNiang — APPROVED** [review]
/approve

File: application/src/test/java/run/halo/app/plugin/DefaultDevelopmentPluginRepositoryTest.java (hunk 1 above)
Comment (guqing) [reply application/src/test/java/run/halo/app/plugin/DefaultDevelopmentPluginRepositoryTest.java:23]: Done!
Very useful advice 🎉

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**